### PR TITLE
Fix for issue #5 - don't require a connection where it's not needed

### LINKED ISSUE #5: Getting the "Failed to retrieve mandatory parameter 'url'." for a build not requiring any Conjur connection
## Summary

We installed and enabled the conjur TC plugin. Now, completely unrelated builds (those not requiring any conjur parameters and not having any Conjur connection configured via their parent project) are failing due to
```
14:34:53 Starting the build on the agent "<agent name>"
14:34:54 Updating tools for build < 1s
14:34:54   Found 1 tool used by the build: maven.3.6.1
14:34:54   All used tools are up-to-date
14:34:55 ERROR: Retrieving conjur agent's shared parameters. Failed to retrieve mandatory parameter 'url'. This should not happen
14:34:55 ERROR: Retrieving conjur agent's shared parameters. Failed to retrieve mandatory parameter 'url'. This should not happen
14:34:55 Stopping build on agent. Reason: fail and stop build command from the agent (ERROR: Retrieving conjur agent's shared parameters. Failed to retrieve mandatory parameter 'url'. This should not happen)
14:34:56 Build was interrupted. Artifacts will not be published for this build
14:34:56 Build finished
```

## Steps to Reproduce
1. Install plugin - no error observed
2. Create a connection in the \<Root project\>/\<ProjectA\>, possibly with wrong setup (note: there's no Test Connection button, unlike in e.g. AWS ECR Connection or in VCS roots) - no error observed in UI or logs
3. Run a build in the \<Root project\>/\<ProjectB\>/\<BuildConfigX\> that needs no Conjur variables nor has any connection set up

## Expected Results
4. See no error

## Actual Results (including error logs, if applicable)
4. The build failed due to the above error

## Reproducible
   * [ ] Always 
   * [ ] Sometimes
   * [ ] Non-Reproducible

We don't know. Probably all the builds would fail but this would affect too many users, so we decided to remove the plugin for now. Will test in more detail on a separate instance again. In any case, even if the Conjur connection (see step 2) was wrong, it should not affect builds from other projects.
   
## Version/Tag number
0.0.1 release of the plugin

## Environment setup
Linux 64bit both server and agent, TC 2021.2.3 version

### PR DESCRIPTION
**Pull request by pe-tr**
### Desired Outcome

Fixes issue #5. 

If there was no Conjur connection defined (on the project level), the builds were failing even though they didn't use any parameters referring to any Conjur instance. As a result, if the Conjur plugin was *enabled*, all the projects (or the _Root) project needed a Conjur connection defined, and all builds were creating a connection when starting, no matter if they needed it or not.

Also note that in some cases the project or TC server administrator does *not* want some projects/builds to access Conjur at all - this fix makes this scenario possible.

### Implemented Changes

No new code was added. Instead, I tried to simplify existing where possible, to provide more descriptive variable names, to make imports more specific, and to get rid of maven warnings.

The crucial fix is the one in the ```ConjurEnvParameterReplacer``` class (previously ```ConjurBuildFeature```) - the check for existence of any Conjur-related parameters in a build is performed *before* trying to access the Conjur connection parameters, not *after* it: here, the ```if {... }``` block is only moved a few lines above.

### Connected Issue/Story

Resolves #5, which is a blocker for even validating the plugin on a subproject on an existing TeamCity instance. Before the fix, the only way to run (or even test) this plugin was to create the connection on the Root project and/or on every single subproject.

#### Changelog

- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] The changes in this PR do not require tests

#### Documentation

- [x] This PR does not require updating any documentation

#### Behavior

- [x] No behavior was changed with this PR (it's just fixing the behavior so it does what the user expects).

#### Security

- [x] There are no security aspects to these changes (strictly speaking, a comment in the code highlights the existence of #7 - fixing it will be the next step)

### CODE REVIEW

**Comment by pe-tr** [comment]
@jtuttle, may I ask for a review this PR please?

**Comment by pe-tr** [comment]
@doodlesbykumbi would you like to take a look at this PR please?

File: teamcity-common/src/main/java/com/cyberark/common/ConjurConnectionParameters.java [hunk 1]
```diff
@@ -15,47 +14,33 @@ public class ConjurConnectionParameters {
...
-        sharedParameters.put(prefix + conjurKeys.getApplianceUrl(), this.getApplianceUrl());
-        sharedParameters.put(prefix + conjurKeys.getAuthnLogin(), this.getAuthnLogin());
-        sharedParameters.put(prefix + conjurKeys.getApiKey(), this.getApiKey());
-        sharedParameters.put(prefix + conjurKeys.getCertFile(), this.getCertFile());
-        sharedParameters.put(prefix + conjurKeys.getFailOnError(), String.valueOf(this.getFailOnError()));
-        sharedParameters.put(prefix + conjurKeys.getVerboseLogging(), String.valueOf(this.getVerboseLogging()));
+        sharedParameters.put(agentParameterPrefix + conjurKeys.getAccount(), this.getAccount());
+        sharedParameters.put(agentParameterPrefix + conjurKeys.getApplianceUrl(), this.getApplianceUrl());
+        sharedParameters.put(agentParameterPrefix + conjurKeys.getAuthnLogin(), this.getAuthnLogin());
+        sharedParameters.put(agentParameterPrefix + conjurKeys.getApiKey(), this.getApiKey()); // TODO !!! this must be added as a password-type param instead, or must not be added at all
```
Comment (infamousjoeg) [inline teamcity-common/src/main/java/com/cyberark/common/ConjurConnectionParameters.java:36]: Is an Issue logged for this TODO item?

**Review by infamousjoeg — APPROVED** [review]
LGTM

File: teamcity-common/src/main/java/com/cyberark/common/ConjurConnectionParameters.java (hunk 1 above)
Comment (pe-tr) [reply teamcity-common/src/main/java/com/cyberark/common/ConjurConnectionParameters.java:36]: yes it is - it's the #7 , which is the next (and probably also the last for near future) on my todo list for this plugin

**Comment by jtuttle** [comment]
@pe-tr Took a quick glance and it looks great, thanks for the fix!

**Pull request merged**